### PR TITLE
Update auth UI to support multiple email factor providers

### PR DESCRIPTION
**Pull request by jaclarke**

### CODE REVIEW

File: edb/server/protocol/auth_ext/ui/components.py [hunk 1]
```diff
@@ -0,0 +1,338 @@
+#
```
Comment (scotttrinh) [inline edb/server/protocol/auth_ext/ui/components.py:1]: I love this whole module ❤️‍🔥

File: edb/server/protocol/auth_ext/_static/interactions.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+      form.action = form.dataset.secondaryAction;
+      setInputValues(hiddenInputs);
+    }
+
+    document
+      .getElementById("show-password-form")
+      .addEventListener("click", () => {
+        moveSliderToIndex(sliderContainer, 1);
+
+        form.action = form.dataset.secondaryAction;
```
[the remark this answers is not in the document]
Comment (jaclarke) [reply edb/server/protocol/auth_ext/_static/interactions.js:38]: Does that work if you don't submit the form using the button, like pressing the Enter key? If you have password + magic link, then both buttons exist, so I'm not sure which action would be used?

File: edb/server/protocol/auth_ext/_static/webauthn-authenticate.js [hunk 3]
```diff
@@ -4,18 +4,24 @@ import {
...
+
 /**
  * Handle the form submission for WebAuthn authentication
  * @param {SubmitEvent} event
  * @param {HTMLFormElement} form
  * @returns void
  */
 export async function onAuthenticateSubmit(event, form) {
-  if (event.submitter?.id !== "webauthn-signin") {
+  if (event.submitter?.id !== "webauthn-signin" || authenticating) {
```
[the remark this answers is not in the document]
Comment (jaclarke) [reply edb/server/protocol/auth_ext/_static/webauthn-authenticate.js:16]: Actually, maybe the `authenticating` check should be after the `event.preventDefault()`?

[the remark this answers is not in the document]
Comment (scotttrinh) [reply edb/server/protocol/auth_ext/_static/webauthn-authenticate.js:16]: Ahh, yes, good catch: we still want to prevent the form from submitting.

File: edb/server/protocol/auth_ext/_static/interactions.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (jaclarke) [reply edb/server/protocol/auth_ext/_static/interactions.js:38]: I can't seem to get that to work. By default it always uses the `formaction` on the first button, even when it's hidden, I guess because it's only visually hidden. Trying to toggle disabled on the hidden buttons, just seems to act like there's no action at all. So it looks like the `formaction` on the hidden buttons would need to be removed when they're hidden, which I feel just brings us full circle.

[the remark this answers is not in the document]
Comment (scotttrinh) [reply edb/server/protocol/auth_ext/_static/interactions.js:38]: Ahh, yeah, if a button has `type="button"` it won't submit at all, which makes sense here. We _could_ use some kind of layout hack to put the buttons in a different order than document-order, but yeah I don't care enough to keep fighting this 😅 

Maybe another approach would be to switch to `fetch` -> `window.location` redirect at some point. My only real issue with the current approach is that it's a little hard to follow the logic unless you read everything altogether and even then I think I'd be prone to introducing subtle bugs here.

Not a blocker, let's not bikeshed on this too much!

**Review by scotttrinh — APPROVED** [review]
This works _really_ well, nice job!
